dags/challenge_dag_factory.py

```diff
@@ -0,0 +1,173 @@
+    dag_params = {
+        "synapse_conn_id": Param(config["synapse_conn_id"], type="string"),
+        "aws_conn_id": Param(config["aws_conn_id"], type="string"),
+        "revision": Param(config["revision"], type="string"),
+        "challenge_profile": Param(config["challenge_profile"], type="string"),
+        "tower_conn_id": Param(config["tower_conn_id"], type="string"),
+        "tower_view_id": Param(config["tower_view_id"], type="string"),
+        "tower_compute_env_type": Param(config["tower_compute_env_type"], type="string"),
+        "bucket_name": Param(config["bucket_name"], type="string"),
+        "key": Param(config["key"], type="string"),
+        "uuid": Param(config["uuid"], type="string"),
+    }
+
+    # Resolve the complete DAG configuration.
+    dag_config = resolve_dag_config(challenge_name, dag_params, config)
+
+    # Create a unique DAG ID for the challenge.
+    dag_id = f"{challenge_name}_challenge_dag"
+
+    @dag(dag_id=dag_id, **dag_config)
+    def challenge_dag():
+        @task
+        def get_new_submissions(**context):
+            hook = SynapseHook(context["params"]["synapse_conn_id"])
+            submissions = hook.ops.get_submissions_with_status(
+                context["params"]["tower_view_id"], "RECEIVED"
+            )
+            return submissions
+
+        @task.branch()
+        def update_submission_statuses(submissions, **context):
+            if submissions:
+                hook = SynapseHook(context["params"]["synapse_conn_id"])
+                for submission in submissions:
+                    hook.ops.update_submission_status(
+                        submission_id=submission, submission_status="EVALUATION_IN_PROGRESS"
+                    )
+                return "stage_submissions_manifest"
+            return "stop_dag"
+
+        @task
+        def stop_dag():
+            # A dummy task to mark the end when there are no submissions.
+            pass
+
+        @task
+        def stage_submissions_manifest(submissions, **context):
+            # Use per-challenge bucket and key settings from the DAG params.
+            bucket_name = context["params"]["bucket_name"]
+            key_root = context["params"]["key"]
+            s3_hook = S3Hook(
+                aws_conn_id=context["params"]["aws_conn_id"],
+                region_name="us-east-1"  # You could also make region configurable.
+            )
+            # Create a CSV manifest from submissions.
+            df = pd.DataFrame({"submission_id": submissions})
+            csv_file = "submissions.csv"
+            df.to_csv(csv_file, index=False)
+            run_uuid = context["params"]["uuid"]
+            # Create a key path that incorporates the unique run identifier.
+            s3_key = f"{key_root}/{run_uuid}/{csv_file}"
+            # Upload the file.
+            s3_hook.load_file(filename=csv_file, key=s3_key, bucket_name=bucket_name)
+            os.remove(csv_file)
```

Comment:
You should not need to write this to a `csv` that is written to disk. I would prefer that you instead load this into an in-memory buffer and load that through the S3 hook. The intention is to prevent files from being written/left there on errors.

See how I am doing it here:

https://github.com/Sage-Bionetworks-Workflows/orca-recipes/blob/be11f73dd718f690309be0f4763405dc511e5dda/dags/synapse-dataset-to-croissant.py#L618-L625


https://github.com/Sage-Bionetworks/synapsePythonClient/blob/c70c634363f0ea06dbf5ecb9c405dc261d9cc242/synapseclient/core/upload/upload_utils.py#L50-L75